reactor-netty-core/src/main/java/reactor/netty/BufferFlux.java

```diff
@@ -0,0 +1,329 @@
+		return handle((b, sink) -> {
+			int readableBytes = b.readableBytes();
+			ByteBuffer copy = b.isDirect() ? ByteBuffer.allocateDirect(readableBytes) : ByteBuffer.allocate(readableBytes);
+			b.copyInto(0, copy, 0, readableBytes);
+			sink.next(copy);
+		});
+	}
+
+	/**
+	 * Convert to a {@literal byte[]} inbound {@link Flux}
+	 *
+	 * @return a {@literal byte[]} inbound {@link Flux}
+	 */
+	public final Flux<byte[]> asByteArray() {
+		return handle((b, sink) -> {
+			int readableBytes = b.readableBytes();
+			byte[] copy = new byte[readableBytes];
+			b.copyInto(0, copy, 0, readableBytes);
+			sink.next(copy);
+		});
+	}
+
+	/**
+	 * Convert to an {@link InputStream} inbound {@link Flux}
+	 * <p><strong>Note:</strong> The underlying {@link Buffer} will be released only when {@link InputStream#close()} is invoked.
+	 * Ensure {@link InputStream#close()} is invoked for any terminal signal:
+	 * {@code complete} | {@code error} | {@code cancel}.
+	 * </p>
+	 *
+	 * @return a {@link InputStream} inbound {@link Flux}
+	 */
+	public final Flux<InputStream> asInputStream() {
+		return handle((b, sink) -> sink.next(new BufferInputStream(b.send())));
+	}
+
+	/**
+	 * Convert to a {@link String} inbound {@link Flux} using the default {@link Charset}.
+	 *
+	 * @return a {@link String} inbound {@link Flux}
+	 */
+	public final Flux<String> asString() {
+		return asString(Charset.defaultCharset());
+	}
+
+	/**
+	 * Convert to a {@link String} inbound {@link Flux} using the provided {@link Charset}.
+	 *
+	 * @param charset the decoding charset
+	 *
+	 * @return a {@link String} inbound {@link Flux}
+	 */
+	public final Flux<String> asString(Charset charset) {
+		Objects.requireNonNull(charset, "charset");
+		return handle((b, sink) -> sink.next(b.readCharSequence(b.readableBytes(), charset).toString()));
+	}
+
+	/**
+	 * Aggregate subsequent byte buffers into a single buffer.
+	 *
+	 * @return {@link BufferMono} of aggregated {@link Buffer}
+	 */
+	@SuppressWarnings("rawtypes")
+	public final BufferMono aggregate() {
+		return Mono.defer(() -> {
+					CompositeBuffer output = CompositeBuffer.compose(alloc);
+					return map(Buffer::send)
+							.collectList()
+							.doOnDiscard(Send.class, Send::close)
+							.handle((list, sink) -> {
+								if (!list.isEmpty()) {
+									for (Send<Buffer> send : list) {
+										output.extendWith(send);
+									}
+								}
+								if (output.readableBytes() > 0) {
+									sink.next(output);
+								}
+								else {
+									sink.complete();
+								}
+							})
+							.doFinally(signalType -> output.close());
+				})
+				.as(BufferMono::maybeFuse);
+	}
+
+	/**
+	 * Returns {@link Flux} of temporary holders of {@link Buffer}s.
+	 * This transfers the ownership to the recipient.
+	 *
+	 * @return {@link Flux} of temporary holders of {@link Buffer}
+	 */
+	public final Flux<Send<Buffer>> send() {
+		return map(Buffer::send);
+	}
+
+	final BufferAllocator alloc;
```

Comment:
nitpick: always feels a bit weird to see fields and constructor after instance methods (asXxx, send...). I know this was the case already in `ByteBufFlux` but unless it was intentional I would consider reordering.

Reply:
we use this formatting everywhere
e.g.
https://github.com/reactor/reactor-netty/blob/main/reactor-netty-core/src/main/java/reactor/netty/transport/TransportConfig.java
https://github.com/reactor/reactor-netty/blob/main/reactor-netty-core/src/main/java/reactor/netty/transport/NameResolverProvider.java

this was done as part of 1.0.x, if somewhere it is not like that I might be an exception